common/src/main/java/com/sequenceiq/cloudbreak/tracing/TracingUtil.java

```diff
@@ -25,6 +29,25 @@
     private TracingUtil() {
     }
 
+    public static Span initSpan(Tracer tracer, String component, String operation) {
+        Span activeSpan = tracer.activeSpan();
+        Span span = tracer.buildSpan(component + " - " + operation)
+                .addReference(References.FOLLOWS_FROM, activeSpan != null ? activeSpan.context() : null)
+                .start();
+        span.setTag(TracingUtil.COMPONENT, component);
+        return span;
+    }
+
+    public static Optional<Span> initOptionalSpan(Tracer tracer, String component, String operation) {
```

Comment:
if I see it well this is almost the same as `initSpan`, except it will return `Optional.empty()` if `tracer.activeSpan()` is `null`
Could you refactor this part to reflect this logic?

Reply:
No, its not worth it